packages/azure-themes/src/azure/IExtendedSemanticColors.ts

```diff
@@ -46,6 +46,8 @@ export interface IExtendedSemanticColors extends ISemanticColors {
   checkBoxCheckHoverTest: string;
   checkBoxCheckedDisabledBackground: string;
   checkBoxDisabled: string;
+  checkBoxIndeterminateDetailsRowCheck: string;
+  checkBoxIndeterminateDetailsRowCheckHover: string;
   checkBoxIndeterminateBackground: string;
   checkBoxIndeterminateDefaultChecked: string;
```

Comment:
The existing standalone checkbox styles slightly varies from the ones used by the DetailsList rows in high contrast themes.

For consistency with what's already being used in the portal (e.g. the knockout detailslist) I introduced two new colors for `checkBoxIndeterminateDetailsRowCheck`/`checkBoxIndeterminateDetailsRowCheckHover` for the DetailsList case.

Existing standalone indeterminate checkbox uses cyan in high contrast:

<img width="226" alt="Screenshot 2024-02-09 at 3 55 24 PM" src="https://github.com/microsoft/fluentui/assets/79171711/5792e160-fc96-485e-9722-bbb165425ca3">

The Portal's details list indeterminate checkbox uses white in high contrast:

<img width="143" alt="Screenshot 2024-02-09 at 3 55 15 PM" src="https://github.com/microsoft/fluentui/assets/79171711/95d3a1a3-efa2-44ef-b03f-98e7a43965f7">